Можно ли встроить скрипт на свой сайт?

sws33 opened the pull request:
Можно ли встроить скрипт на свой сайт, чтобы youtube-видео на сайте включились сразу с переводом?

SenkHub commented:
Применим ли данный метод перевода для не русифицированных игр, если делать захват звука с карты?

ilyhalight committed:
Merge pull request #831 from ilyhalight/vot.js1.3.0

Update to vot.js 1.3.0

ilyhalight committed:
updated to vot.js 1.3.1

ilyhalight committed:
Merge pull request #832 from ilyhalight/vot.js1.3.1

Updated to vot.js 1.3.1

ilyhalight committed:
fix incorrect checkbox name

ilyhalight committed:
Merge pull request #836 from ilyhalight/fix-invalid-audiobooster-name

Fix incorrect checkbox name

ilyhalight committed:
fix multimedia keys if browser support audio context

ilyhalight committed:
Merge pull request #844 from ilyhalight/fix-multimedia-keys

fix multimedia keys if browser support audio context

ilyhalight committed:
added phrases for new checkbox

ilyhalight committed:
fix bypass mediacsp checkbox states on activate restore multimedia keys checkbox

ilyhalight committed:
fix audio player speed but audio glitched

ilyhalight committed:
updated gm values

ilyhalight committed:
rewrited with typescript

ilyhalight committed:
added some typings

ilyhalight committed:
added typescript support

ilyhalight committed:
fix typings

ilyhalight committed:
version bumped to 1.7.1-beta2

ilyhalight committed:
debug.js renamed to debug.ts

ilyhalight committed:
added decibels-percentage converter

ilyhalight committed:
moved initAudioContext to player.ts

ilyhalight committed:
rewrited audio player logic

ilyhalight committed:
updated changelog

ilyhalight committed:
updated VideoHandler typings

ilyhalight committed:
updated to vot.js 1.3.2

ilyhalight committed:
fix miss, porntn domain replaced to ricktube

ilyhalight committed:
fix invalid sequence

ilyhalight committed:
added missed comma

ilyhalight committed:
removed incestflix

ilyhalight committed:
updated to vot.js 1.3.3

ilyhalight committed:
updated changelog.md

ilyhalight committed:
fix bilibili url

ilyhalight committed:
set match by coursehunterLike array

ilyhalight committed:
updated to vot.js 1.3.4 (added incestflix #705)

ilyhalight committed:
added blacklist

ilyhalight committed:
updated docs

ilyhalight committed:
fix some visual issues

ilyhalight committed:
added debug logs

ilyhalight committed:
added sub checkbox style

ilyhalight committed:
updated new player checkbox logic

ilyhalight committed:
updated docs

NullVerdict committed:
Перенос Tone.js и исправление громкости

1. Переместил Tone.js в headers.js для уменьшения общего веса скрипта.
2. Перенес this.volumeOnStart для корректного восстановления громкости при отключении перевода на стримах.
3. Убрал Promise.all, так как в данном контексте он лишний.

Примечание: был обнаружен баг с трансляциями — при увеличении скорости видео аудио исчезает.

ilyhalight committed:
Merge pull request #862 from SashaXser/infdev

Перенос Tone.js и исправление громкости

ilyhalight committed:
fix typings for tone after #862

ilyhalight committed:
removed test comment

ilyhalight committed:
updated

ilyhalight committed:
removed dom-parser from porntn helper

ilyhalight committed:
Fixed long waiting for translation for new translation requests for YouTube (status = 6) (#1#issuecomment-2433274910, #868#issuecomment-2436080833)

ilyhalight committed:
builded without dev mode

ilyhalight committed:
added missed phrase

ilyhalight committed:
Merge pull request #869 from ilyhalight/fix-audio-player

Rewrited audio player logic with Tone.js (temporary)

ilyhalight committed:
changed audio context player tone.js -> chaimu

ilyhalight committed:
Merge pull request #874 from ilyhalight/fix-audio-player

Changed audio context player Tone.js --> chaimu

ilyhalight committed:
fix translation button on change video with old player

ilyhalight committed:
Merge pull request #879 from ilyhalight/fix-audio-player

Fix translation button on change video with old player

ilyhalight committed:
updated vot.js to 1.3.8

ilyhalight committed:
updated depends

ilyhalight committed:
ignore test file

ilyhalight committed:
fix some errors in changelog

ilyhalight committed:
added guide how to update vot.js

ilyhalight committed:
added rounding in secsToStrTime func

ilyhalight committed:
migrated from import to use

ilyhalight committed:
updated version

ilyhalight committed:
added support view video in client player (without open public link, only for /client/disk)

ilyhalight committed:
revert postinstall

ilyhalight committed:
eslint-plugin-sonarjs was removed due to too strong an impact on build performance and too low an impact

ilyhalight committed:
Merge pull request #892 from ilyhalight/dev

Update 1.7.1

ilyhalight committed:
update vot.js to 1.3.9

ilyhalight committed:
fix lightnincss on linux

ilyhalight committed:
Merge pull request #895 from ilyhalight/dev

update vot.js to 1.3.9

ilyhalight committed:
added the ability to enable use an experimental variation of Yandex voices for some videos

ilyhalight committed:
Merge pull request #898 from ilyhalight/dev

Option to use an experimental variation of Yandex voices for some videos

ilyhalight committed:
added cache bypass on force update localization files

ilyhalight committed:
added cache bypass on force update localization files

ilyhalight committed:
Merge branch 'dev' of https://github.com/ilyhalight/voice-over-translation into dev

ilyhalight committed:
removed todo comment

ilyhalight committed:
Merge pull request #901 from ilyhalight/dev

added cache bypass on force update localization files

ilyhalight committed:
updated chaimu to 1.0.3

ilyhalight committed:
Merge pull request #903 from ilyhalight/dev

Updated chaimu to 1.0.3

ilyhalight committed:
fix yandex disk /i/ (#908)

ilyhalight committed:
Merge pull request #910 from ilyhalight/dev

Fix yandex disk /i/ path

NullVerdict committed:
Доработка VideoObserver под ShadowDOM

Поддержка Shadow DOM:
    Переопределена функция attachShadow для автоматического поиска видео в новых теневых корнях.
    Добавлен метод observeShadowRoot для наблюдения за изменениями в Shadow DOM, чтобы находить новые видеоэлементы.

(Нужно доработать findContainer)

ilyhalight committed:
Merge pull request #914 from SashaXser/dev

Доработка ShadowDOM в VideoObserver

ilyhalight committed:
added shadow root climb

ilyhalight committed:
added support main reddit frontend (#732)

ilyhalight committed:
updated vot.js to 1.4.0

ilyhalight committed:
added set url property for GM_fetch

ilyhalight committed:
fix porntn

ilyhalight committed:
updated vot.js to 1.4.1

NullVerdict committed:
Замена removeEventListener на AbortController

Заменена логика удаления обработчиков событий с removeEventListener на использование AbortController для более удобного и эффективного управления отменой событий.

ilyhalight committed:
migrated from vot.js to @vot.js/ext and @vot.js/shared

NullVerdict committed:
Удаление избыточного кода

ilyhalight committed:
Merge pull request #927 from SashaXser/dev

 Небольшие правки №2

NullVerdict committed:
Изменение проверки недоступности сервера Яндекса

NullVerdict committed:
Доработка проверки недоступности сервера Яндекса

ilyhalight committed:
Merge pull request #928 from SashaXser/dev

Изменение проверки недоступности сервера Яндекса

NullVerdict committed:
Исправление #928

1. Исправлены ошибки, связанные с this.translateProxyEnabled.
2. Улучшена проверка готовности видео и удален лишний код в методе searchInRoot.

ilyhalight committed:
Merge pull request #930 from SashaXser/dev

Исправление #928

ilyhalight committed:
added vkvideo.ru to match domains (#931)

ilyhalight committed:
updated vot.js to 2.0.4

ilyhalight committed:
builded & updated

ilyhalight committed:
updated vot.js to 2.0.5

ilyhalight committed:
added @toil/translate for foswly translate typings

ilyhalight committed:
updated depends

ilyhalight committed:
converter if's logic moved to convertData func

ilyhalight committed:
updated 'yandex' proxy backend to foswly multi-providers backend

ilyhalight committed:
rewritted storage logic to typescript

ilyhalight committed:
updated some default values

ilyhalight committed:
builded

ilyhalight committed:
renamed 'Translation service' -> 'Error translation service' and 'Detect service' to 'Language detection service' to avoid confusion

ilyhalight committed:
updated features

ilyhalight committed:
updated sites

ilyhalight committed:
change yt autotranslated subtitles lang to response lang

ilyhalight committed:
removed whitelist for dont translate your lang

ilyhalight committed:
added youtube player types

ilyhalight committed:
builded and updated

ilyhalight committed:
added format param for subtitles

ilyhalight committed:
added getVideoData for VK (#770, #932)

ilyhalight committed:
Create FUNDING.yml

ilyhalight committed:
moved to vot.js

ilyhalight committed:
added translation delayed message on get 5 repeated 'translation take about a minute' message (#919)

ilyhalight committed:
updated locales

ilyhalight committed:
builded and updated

ilyhalight committed:
fix epicgames (#944)

ilyhalight committed:
added support cloudflare streams (#941)

ilyhalight committed:
fix linkedin learning with logined (#882)

ilyhalight committed:
added douyin support (#876) and small update getVideoData logic

ilyhalight committed:
updated translationHelp destructurization

ilyhalight committed:
added send notification on translation completed (#934)

ilyhalight committed:
fix xvideos mobile player conflict (#866)

ilyhalight committed:
fix mobile vk (#957)

ilyhalight committed:
version bump

ilyhalight committed:
Merge pull request #961 from ilyhalight/dev

1.8.0

ilyhalight committed:
removed notification if translation exists

ilyhalight committed:
feat: add loom and alt bannedvideo domain (#946, #34)

ilyhalight committed:
chore(deps): bump vot.js version to 2.0.16

ilyhalight committed:
docs: add reference issue for vkvideo

ilyhalight committed:
chore: version bump to 1.8.1

ilyhalight committed:
Merge pull request #972 from ilyhalight/dev

Update 1.8.1

NullVerdict committed:
Не переводить с родного языка" для 2-х и более языков #818

1. Реализован мульти-выбор для функции createVOTSelect.
2. Добавлена возможность выбора двух и более языков в параметре "Не переводить с родного языка".

ilyhalight committed:
Merge pull request #974 from SashaXser/dev

Не переводить с родного языка" для 2-х и более языков #818

NullVerdict committed:
Доработка субтитров и исправление бага в createVOTSelect

1. Реорганизовал функции для обработки субтитров, выделив их в отдельный класс SubtitlesProcessor.
2. Исправил баг, из-за которого при выборе субтитров не обновлялся заголовок и не происходило выделение в меню.
3. Изменил логику активации проксирования субтитров
4. Убрал возможность отключения языка браузера при выборе родного языка

NullVerdict committed:
Исправление выбора родного языка

Изменил проверку с языка браузера на проверку наличия хотя бы одного активного языка.

ilyhalight committed:
Merge pull request #975 from SashaXser/dev

Доработка субтитров и исправление бага в createVOTSelect

ilyhalight committed:
docs: updated changelog

ilyhalight committed:
fix: removed show download button for streams

ilyhalight committed:
fix: activating audio player with streams

ilyhalight committed:
feat: add artstation

ilyhalight committed:
chore(deps): update vot.js to 2.0.17

ilyhalight committed:
docs: update changelog

ilyhalight committed:
build: new version

ilyhalight committed:
docs: small improve

ilyhalight committed:
chore: update vot.js to 2.0.19

NullVerdict committed:
Поддержка тачскрина

1. Реализована возможность перемещения кнопки и субтитров на мобильных устройствах.
2. Небольшое упрощение кода..

ilyhalight committed:
Merge pull request #978 from SashaXser/dev

Поддержка тачскрина

ilyhalight committed:
fix: validate audio url with translateProxyEnable=2 (#976)

NullVerdict committed:
Исправление #978

На мобильных устройствах возникали проблемы с нажатием на кнопку, но теперь все должно работать корректно.

ilyhalight committed:
Merge pull request #980 from SashaXser/dev

Исправление #978

ilyhalight committed:
Merge branch 'dev' into test-976

ilyhalight committed:
Merge pull request #982 from ilyhalight/test-976

Fix #976

ilyhalight committed:
chore: update vot.js to 2.1.0

ilyhalight committed:
chore: update getEventContainer logic for twitter and small adaptive for vot.js 2.1.0

ilyhalight committed:
build: dev version

ilyhalight committed:
docs: update changelog

ilyhalight committed:
chore: update vot.js to 2.1.1

ilyhalight committed:
fix: gm_info polyfill proxying

ilyhalight committed:
fix: enable proxying for userscripts extension

ilyhalight committed:
chore(deps): update vot.js to 2.1.2

ilyhalight committed:
feat: add douyin feed support (#876)

ilyhalight committed:
chore: bump version

ilyhalight committed:
Merge pull request #993 from ilyhalight/dev

Update 1.8.2

ilyhalight committed:
docs: generated new sites lits

ilyhalight committed:
chore: add set option value param

ilyhalight committed:
fix: volume sliders (#996) and add missed dontTranslateLanguage converter (#995, #997)

ilyhalight committed:
build: new version

ilyhalight committed:
chore: version bump

ilyhalight committed:
Merge pull request #998 from ilyhalight/dev

Update 1.8.3

ilyhalight committed:
fix: added missed types

ilyhalight committed:
fix: block swiping if dragging false (#1006)

ilyhalight committed:
fix: hide button on mobile (#1005)

ilyhalight committed:
style: fix old issue with prettier

ilyhalight committed:
build: dev version

ilyhalight committed:
docs: update changelog

ilyhalight committed:
chore: fix prettier for editors autosave

ilyhalight committed:
feat: add bug report button to menu

ilyhalight committed:
chore: bump version

ilyhalight committed:
Merge pull request #1007 from ilyhalight/dev

Update 1.8.4

ilyhalight committed:
chore(deps): bump vot.js version

ilyhalight committed:
build: version with fix #1011

ilyhalight committed:
feat: added support all yandex disk tlds (#1012)

ilyhalight committed:
chore(deps): bump vot.js version

ilyhalight committed:
docs: update changelog

ilyhalight committed:
build: dev version

NullVerdict committed:
Доработка GM_Fetch для Violentmonkey

Теперь можно отправлять запросы напрямую в Яндекс, без использования прокси. С Greasemonkey возникает проблема — по какой-то причине удаляются заголовки sec. Можно протестировать и доработать решение для других расширений.

ilyhalight committed:
Merge pull request #1018 from SashaXser/dev

Доработка GM_Fetch для Violentmonkey

NullVerdict committed:
Небольшое упрощение кода

ilyhalight committed:
Merge pull request #1027 from SashaXser/dev

Небольшое упрощение кода

ilyhalight committed:
chore: added rt news and some xvideos domains to match list (#1015)

ilyhalight committed:
chore(deps): update vot.js to 2.1.5

ilyhalight committed:
build: updated version

ilyhalight committed:
chore: proxying by country -> proxying if yandex.ru isn't reach

ilyhalight committed:
build: rebuild dev version

NullVerdict committed:
Исправление парсинга заголовков  GM_Fetch

Конструктор Headers не позволил добавить некоторые заголовки, поэтому я добавил фильтрацию, чтобы избежать ошибок.

ilyhalight committed:
Merge pull request #1029 from SashaXser/dev

Исправление парсинга заголовков  GM_Fetch

ilyhalight committed:
chore: bump vot.js to 1.2.6

ilyhalight committed:
build: updated version

ilyhalight committed:
git: fix conflict

ilyhalight committed:
revert: remove response url from #1027

ilyhalight committed:
docs: updated changelog

ilyhalight committed:
Revert "build: rebuild dev version"

This reverts commit 6fab8c056e58510bf577d3ee1d35ccfc576f3a21.

ilyhalight committed:
Revert "chore: proxying by country -> proxying if yandex.ru isn't reach"

This reverts commit a599bdd28fcf2ea1915e147199d6b850ad6c867d.

ilyhalight committed:
build: rebuild after reverts

ilyhalight committed:
chore: added selected value to onSelectCB in createVOTSelect

ilyhalight committed:
chore: added logic to set proxy status

ilyhalight committed:
chore: added missed phrase

ilyhalight committed:
chore: move ?? condition to default subtitles value

ilyhalight committed:
refactor: removed extra await

ilyhalight committed:
chore(deps): bump vot.js to 1.2.7

ilyhalight committed:
refactor: youtubeUtils -> YouTubeHelper

ilyhalight committed:
build: updated version

ilyhalight committed:
docs: updated changelog

ilyhalight committed:
fix: get detected language

NullVerdict committed:
Упрощение парсинга заголовков GM_fetch

1. Убрал преобразование в Headers, так как оно не имело смысла.
2. Улучшил парсинг заголовков с помощью flatMap и exec (возможно, можно ещё упростить).
3. Исправил проблему с Set-Cookie (нужно протестировать).

ilyhalight committed:
Merge pull request #1035 from SashaXser/dev

Упрощение парсинга заголовков GM_fetch

ilyhalight committed:
chore(deps): bump vot.js to 2.1.8

ilyhalight committed:
chore: added support coursera preview

ilyhalight committed:
build: new version

NullVerdict committed:
Fix #1014

ilyhalight committed:
Merge branch 'SashaXser-dev' into dev

ilyhalight committed:
build: rebuild

ilyhalight committed:
chore: moved l10n-tui logic to separated util

ilyhalight committed:
chore: small fixes after migration to localize-tui

NullVerdict committed:
Упрощение VideoObserver

1. Упрощен поиск видео
2. Упрощена проверка готовности видео
3. Улучшена читаемость кода

NullVerdict committed:
Кэширование видеоэлементов в VideoObserver

NullVerdict committed:
Fix checkVideoState

ilyhalight committed:
Merge pull request #1038 from SashaXser/dev

Упрощение VideoObserver

NullVerdict committed:
Доработка #1038

1. Оптимизирована обработка DOM: элементы теперь собираются в массив и обрабатываются итеративно, что уменьшает глубину рекурсивных вызовов.
2. Введена буферизация изменений: MutationObserver накапливает узлы в Set и обрабатывает их пачкой в requestIdleCallback, снижая нагрузку на процессор.
3. Исправлено дублирование кнопки на Reddit.

ilyhalight committed:
Merge pull request #1042 from SashaXser/dev

Доработка #1038

ilyhalight committed:
fix: epicgames multi iframes (#1043)

ilyhalight committed:
chore: updated vot.js to 2.1.9

ilyhalight committed:
refactor: simplify find iframe

ilyhalight committed:
docs: updated changelog

ilyhalight committed:
chore(deps): updated vot.js to 2.1.10

ilyhalight committed:
feat: added support bitview (#1041)

ilyhalight committed:
chore: updated vot.js to 2.1.10

ilyhalight committed:
chore: bump version

ilyhalight committed:
Merge pull request #1051 from ilyhalight/dev

Update 1.8.5

ilyhalight committed:
chore: updated for vot.js 2.1.11+

ilyhalight committed:
chore(deps): bump vot.js to 2.1.12

ilyhalight committed:
build

ilyhalight committed:
docs: fix typo

ilyhalight committed:
Merge pull request #1057 from ilyhalight/dev

Fix YT errors with GreasyMonkey
